src/libguarded/guarded.hpp

```diff
@@ -19,6 +19,48 @@
 namespace libguarded
 {
 
+template <typename T, typename D>
+struct Handle
+{
+    Handle(T* t, D& d) :
+        ref(*t),
+        deleter(std::move(d)) {}
+
+    Handle(const Handle& handle) = delete;
+    Handle(Handle&& handle) :
+        ref(handle.ref),
+        deleter(std::move(handle.deleter)) {}
+
+    ~Handle()
+    {
+        deleter(&ref);
+    }
+
+    T* operator->() const
+    {
+        return &ref;
+    }
+
+    const T& operator*() const
+    {
+        return *ref;
```

Comment:
there should be no dereferencing here: you are dereferencing T, not T*. The body should be identical to the non const overload below